cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -792,6 +792,112 @@ mod tests {
         assert_build_file(&env_path, &package_name, &file_name, &file_content);
     }
 
+    #[test]
+    fn build_wraps_binaries_with_preserved_arg0() {
+        let package_name = String::from("foo");
+        let file_name = String::from("print_arg0");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [install]
+            go.pkg-path = "go"
+
+            [build.{package_name}]
+            command = """
+                go build main.go
+                mkdir -p $out/bin
+                cp main $out/bin/{file_name}
+            """
+        "#};
+
+        let (mut flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        let arg0_code = indoc! {r#"
+            package main
+
+            import (
+                "fmt"
+                "os"
+            )
+
+            func main() {
+                fmt.Println(os.Args[0])
+            }
+        "#};
+        fs::write(env_path.join("main.go"), arg0_code).unwrap();
+
+        if let Client::Mock(ref mut client) = flox.catalog_client {
```

Comment:
Just FYI you can use a `let ... else ... ` binding for this rather than `if let ...`. It's a little more concise and meant for cases like this where the destructuring is meant to succeed or panic.

Reply:
This is already copy/pasted several places in the codebase so I'm going to leave it for now.